hexa/databases/tests/test_utils.py

```diff
@@ -0,0 +1,137 @@
+            cls.USER_SABRINA,
+            name="Test Workspace",
+            description="Test workspace",
+            countries=[],
+        )
+
+        cls.WORKSPACE = Workspace.objects.create_if_has_perm(
+            cls.USER_SABRINA,
+            name="Test Workspace",
+            description="Test workspace",
+            countries=[],
+        )
+        setattr(cls.WORKSPACE, "database", cls.DB1)
+        WorkspaceMembership.objects.create(
+            user=cls.USER_SABRINA,
+            workspace=cls.WORKSPACE,
+            role=WorkspaceMembershipRole.VIEWER,
+        )
+
+    @mock.patch("psycopg2.connect")
+    @mock.patch("hexa.databases.utils.Database")
+    def test_get_database_tables_empty(self, mock_connect, mock_db):
+        mock_db.get.return_value = self.DB1
+        mock_connection = mock_connect.return_value
+        mock_context_object = mock_connection.__enter__.return_value
+        cursor = mock_context_object.cursor.return_value.__enter__.return_value
+        cursor.fetchall.return_value = []
+
+        result = get_database_definition(self.WORKSPACE)
+        self.assertEqual(0, result["total_items"])
+
+    @mock.patch("psycopg2.connect")
+    def test_get_database_tables(self, mock_connect):
+        table_name = "database_tutorial"
+        row_count = 2
+        table = {"table_name": table_name, "columns": [], "row_count": row_count}
+        sample = [{"name": "test", "value": "test"}]
+
+        mock_connection = mock_connect.return_value
+        mock_context_object = mock_connection.__enter__.return_value
+        cursor = mock_context_object.cursor.return_value.__enter__.return_value
+        cursor.fetchall.return_value = [dictrow_from_dict(table)]
+        cursor.fetchone.return_value = {"row_count": 2}
+
+        with mock.patch("hexa.databases.utils.Database") as mock_db:
+            mock_db.get.return_value = self.DB1
+            with mock.patch(
+                "hexa.databases.utils.get_table_data"
+            ) as mocked_get_table_data:
+                mocked_get_table_data.return_value = sample
+                result = get_database_definition(self.WORKSPACE)
+
+        self.assertEqual(1, result["total_items"])
+        self.assertEqual(table_name, result["items"][0]["name"])
+
+    @mock.patch("psycopg2.connect")
+    @mock.patch("hexa.databases.utils.Database")
+    def test_get_table_definition_not_found(self, mock_connect, mock_db):
+        mock_db.get.return_value = self.DB1
+        mock_connection = mock_connect.return_value
+        mock_context_object = mock_connection.__enter__.return_value
+        cursor = mock_context_object.cursor.return_value.__enter__.return_value
+        cursor.fetchall.return_value = []
+
+        result = get_database_definition(self.WORKSPACE)
+        self.assertEqual(0, result["total_items"])
+
+    @mock.patch("psycopg2.connect")
+    def test_get_table_definition(self, mock_connect):
```

Comment:
It would be interesting to check what happens if you try to get details of a non-existent table.